systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlConfigurationST.java

```diff
@@ -225,6 +237,81 @@ void testConfigurationFileIsCreated() {
+                            CruiseControlServerParameters.CONCURRENT_LEADER_MOVEMENTS.getDefaultValue())
+                    .addToConfig(CruiseControlServerParameters.REPLICATION_THROTTLE.getName(),
+                            CruiseControlServerParameters.REPLICATION_THROTTLE.getDefaultValue())
+                .build();
+
+            kafka.getSpec().setCruiseControl(cruiseControl);
+        });
+
+        LOGGER.info("Verifying that CC pod is rolling, after changing options");
+        DeploymentUtils.waitTillDepHasRolled(CruiseControlResources.deploymentName(CLUSTER_NAME), 1, cruiseControlSnapShot);
+
+        LOGGER.info("Verifying that Kafka pods did not roll");
+        StatefulSetUtils.waitForNoRollingUpdate(KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), kafkaSnapShot);
+
+        LOGGER.info("Verifying new configuration in the Kafka CR");
+        Pod cruiseControlPod = kubeClient().listPodsByPrefixInName(CRUISE_CONTROL_POD_PREFIX).get(0);
+        String cruiseControlPodName = cruiseControlPod.getMetadata().getName();
+
+        // Get CruiseControl resource properties
+        for (Container container : cruiseControlPod.getSpec().getContainers()) {
+            if (container.getName().equals("cruise-control")) {
+                cruiseControlContainer = container;
+            }
+        }
+        for (EnvVar envVar : Objects.requireNonNull(cruiseControlContainer).getEnv()) {
+            if (envVar.getName().equals("CRUISE_CONTROL_CONFIGURATION")) {
+                cruiseControlConfiguration = envVar;
+            }
+        }
+        InputStream configurationContainerStream = new ByteArrayInputStream(Objects.requireNonNull(cruiseControlConfiguration).getValue().getBytes(StandardCharsets.UTF_8));
+        Properties containerConfiguration = new Properties();
+        containerConfiguration.load(configurationContainerStream);
+
+        String configurationFileContent = cmdKubeClient().execInPod(cruiseControlPodName, "/bin/bash", "-c", "cat " + CRUISE_CONTROL_CONFIGURATION_FILE_PATH).out();
+        InputStream configurationFileStream = new ByteArrayInputStream(configurationFileContent.getBytes(StandardCharsets.UTF_8));
+        Properties fileConfiguration = new Properties();
+        fileConfiguration.load(configurationFileStream);
+
+        LOGGER.info("Verifying Cruise control performance options are set in Kafka CR");
+        List<String> checkCCProperties = Arrays.asList(
+                CruiseControlServerParameters.CONCURRENT_INTRA_PARTITION_MOVEMENTS.getName(),
+                CruiseControlServerParameters.CONCURRENT_PARTITION_MOVEMENTS.getName(),
+                CruiseControlServerParameters.CONCURRENT_LEADER_MOVEMENTS.getName(),
+                CruiseControlServerParameters.REPLICATION_THROTTLE.getName()
+        );
+
+        LOGGER.info("Verifying Cruise control configuration keys changes");
+        IsIterableContainingInAnyOrder.containsInAnyOrder(containerConfiguration.stringPropertyNames(),
```

Comment:
Because you're not using assertThat I think this doesn't error if it doesnt match. I also recommend you static import the matches to reduce the length of the line.

Reply:
Fixed by simplifying this whole part. Thank you for looking at it. I hope now it's better and simpler. :)